### PR TITLE
[DRAFT_PR_IGNORE_IT]Destination-Snowflake: added basic part for support oauth login mode (new PR https://github.com/airbytehq/airbyte/pull/11093)

### PR DESCRIPTION
**Pull request by etsybaev**
new PR https://github.com/airbytehq/airbyte/pull/11093

### CODE REVIEW

**Comment by etsybaev** [comment]
https://github.com/dbt-labs/dbt-core/pull/2141/files/d5c24b0c6314ab5dab75f63bedb667087b756afc#diff-aad1789413264edc81d5e71c0aa3d6bec8a42b01b29e3d5810c6b0ce21738226

**Comment by etsybaev** [comment]
Proceeding with normalization https://github.com/airbytehq/airbyte/pull/11093

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java [hunk 1]
```diff
@@ -4,58 +4,189 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * SnowflakeDatabase contains helpers to create connections to and run queries on Snowflake.
  */
 public class SnowflakeDatabase {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeDatabase.class);
+  private static final int PAUSE_BETWEEN_TOKEN_REFRESH_MIN = 7;
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java:39]: how come this value?

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java [hunk 2]
```diff
@@ -4,58 +4,189 @@
...
+      .connectTimeout(Duration.ofSeconds(10))
+      .build();
+
   private static DataSource createDataSource(final JsonNode config) {
-    final String host = config.get("host").asText();
-    final String username = config.get("username").asText();
-    final String password = config.get("password").asText();
+    final HikariDataSource dataSource = new HikariDataSource();
+
+    final StringBuilder jdbcUrl = new StringBuilder(String.format("jdbc:snowflake://%s/?",
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java:55]: it seems like "jdbc:snowflake://%s/?" can be moved to the static variable

Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java:55]: it seems like "jdbc:snowflake://%s/?" can be moved to the static variable

**Comment by alexandr-shegeda** [comment]
@etsybaev could you possibly update the PR description?
- user impact
- recommended reading order

**Comment by etsybaev** [comment]
New PR https://github.com/airbytehq/airbyte/pull/11093

**Pull request closed without merging**